pkgs/by-name/ni/niri/package.nix

```diff
@@ -38,56 +47,76 @@ rustPlatform.buildRustPackage rec {
     };
   };
 
+  strictDeps = true;
+
   nativeBuildInputs = [
+    clang
     pkg-config
     rustPlatform.bindgenHook
-    autoPatchelfHook
-    clang
-  ];
-
-  buildInputs = [
-    wayland
-    systemd # For libudev
-    seatd # For libseat
-    libdisplay-info
-    libxkbcommon
-    libinput
-    mesa # For libgbm
-    fontconfig
-    stdenv.cc.cc.lib
-    pipewire
-    pango
   ];
 
-  runtimeDependencies = [
-    wayland
-    mesa
-    libglvnd # For libEGL
-  ];
+  buildInputs =
+    [
+      libdisplay-info
+      libglvnd # For libEGL
+      libinput
+      libxkbcommon
+      mesa # For libgbm
+      pango
+      seatd
+      wayland # For libwayland-client
+    ]
+    ++ lib.optional (withDbus || withScreencastSupport || withSystemd) dbus
+    ++ lib.optional withScreencastSupport pipewire
+    ++ lib.optional withSystemd systemd; # Also includes libudev
 
-  passthru.providedSessions = [ "niri" ];
+  buildFeatures =
+    lib.optional withDbus "dbus"
+    ++ lib.optional withDinit "dinit"
+    ++ lib.optional withScreencastSupport "xdp-gnome-screencast"
+    ++ lib.optional withSystemd "systemd";
+  buildNoDefaultFeatures = true;
 
-  postPatch = ''
-    patchShebangs ./resources/niri-session
-    substituteInPlace ./resources/niri.service \
-      --replace-fail '/usr/bin' "$out/bin"
-  '';
+  postInstall =
+    ''
+      install -Dm0644 resources/niri.desktop -t $out/share/wayland-sessions
+    ''
+    + lib.optionalString withSystemd ''
+      install -Dm0755 resources/niri-session -t $out/bin
+      install -Dm0644 resources/niri{-shutdown.target,.service} -t $out/share/systemd/user
+      install -Dm0644 resources/niri-portals.conf -t $out/share/xdg-desktop-portal
```

Comment:
Is xdg-desktop-portal inherently tied to systemd? I mean, NixOS *is* a systemd distro, so it's a very esoteric setup to ever use this package on non-NixOS with non-systemd init system, with xdg desktop portals. but afaik nixpkgs is meant to in general accomodate non-systemd init systems. wouldn't this include xdg-desktop-portal support? or are xdg-desktop-portals actually just inherently systemd-based such that those setups have little use for this file?

Reply:
> Is xdg-desktop-portal inherently tied to systemd?

There is a close tie, but AFAIK it *might* be able to run with just dbus -- so it'd probably be better gated by `withDbus`